Src/Web/Web.Shared.Net/RequestTrackingTelemetryModule.cs

```diff
@@ -309,6 +337,50 @@ internal void OverrideCorrelationIdLookupHelper(CorrelationIdLookupHelper correl
         }
 
         /// <summary>
+        /// A request must be tracked as Active in order for telemetry to be recorded within OnEndRequest().
+        /// <see cref="System.Web.Handlers.TransferRequestHandler">
+        /// TransferRequestHandler can create a Child request to route extension-less requests to a controller.
+        /// (ex: site/home -> site/HomeController.cs)
+        /// We do not want duplicate telemetry logged for both the Parent and Child requests, so the activeRequests will be created OnBeginRequest.
+        /// When the child request OnEndRequest, the id will be removed from this dictionary and telemetry will not be logged for the parent.
+        /// </see>
+        /// </summary>
+        /// <remarks>
+        /// Unit test projects cannot create an [internal] IIS7WorkerRequest object.
+        /// Without this object, we cannot modify the Request.Headers without throwing a PlatformNotSupportedException.
+        /// Unit tests will have to initialize the RequestIdHeader.
+        /// The second IF will ensure the id is added to the activeRequests.
+        /// </remarks>
+        private void EnsureIsActiveRequestSafe(HttpContext context)
```

Comment:
I don't understand how this method is safe. It still attempts to write to the headers